### PR TITLE
feat: add TJ_AVAX_USDC_AUTO token

### PR DESCRIPTION
**Pull request by cehali**

### CODE REVIEW

File: packages/oracle-node/src/fetchers/evm-chain/avalanche/evm-fetcher/sources/trader-joe-auto/TraderJoeAutoRequestHandlers.ts [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+import { MulticallParsedResponses } from "../../../../../../types";
+import { TEN_AS_BASE_OF_POWER } from "../../../../shared/contants";
+import { getTokensPricesFromLocalCache } from "../../../../shared/utils/get-tokens-prices-from-local-cache";
+
+export type TraderJoeAutoPoolTokensDetailsKeys =
+  keyof typeof traderJoeAutoPoolTokenContractDetails;
+
+const FIRST_TOKEN_INDEXES_FROM_CONTRACT_RESPONSE = [0, 66];
+const SECOND_TOKEN_INDEXES_FROM_CONTRACT_RESPONSE = [66, 130];
+const DEFAULT_DECIMALS = 18;
```
Comment (koonopek) [inline packages/oracle-node/src/fetchers/evm-chain/avalanche/evm-fetcher/sources/trader-joe-auto/TraderJoeAutoRequestHandlers.ts:15]: Maybe we could rename to `TraderJoePrecision` or `TraderJoeDecimals`

File: packages/oracle-node/src/fetchers/evm-chain/avalanche/evm-fetcher/sources/trader-joe-auto/TraderJoeAutoRequestHandlers.ts [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+      extractValueFromMulticallResponse(response, address, "totalSupply")
+    );
+    const totalSupplyDecimals = new Decimal(
+      extractValueFromMulticallResponse(response, address, "decimals")
+    ).toNumber();
+    return this.serializeDecimals(totalSupply, totalSupplyDecimals);
+  }
+
+  serializeDecimals(balance: Decimal, tokenDecimals: number) {
+    const serializedDecimals = DEFAULT_DECIMALS - tokenDecimals;
```
Comment (koonopek) [inline packages/oracle-node/src/fetchers/evm-chain/avalanche/evm-fetcher/sources/trader-joe-auto/TraderJoeAutoRequestHandlers.ts:107]: Maybe we should add assert that `seralizedDecimals > 0`? Or maybe not ;)

File: packages/oracle-node/test/fetchers/avalanche-evm.spec.ts [hunk 3]
```diff
@@ -371,6 +372,45 @@ describe("Avalanche EVM fetcher", () => {
...
+        { mainProvider: provider },
+        multicallContract.address,
+        requestHandlers
+      );
+
+      await saveMockPricesInLocalDb([11.65, 1], ["AVAX", "USDC"]);
+
+      const result = await fetcher.fetchAll(["TJ_AVAX_USDC_AUTO"]);
+      expect(result).toEqual([
+        { symbol: "TJ_AVAX_USDC_AUTO", value: 0.7906939799374457 },
```
[the remark this answers is not in the document]
Comment (cehali) [reply packages/oracle-node/test/fetchers/avalanche-evm.spec.ts:409]: Why do you think it is weird? This implementation and results were consulted with the DeltaPrime team and they were okay with that.

File: packages/oracle-node/src/fetchers/evm-chain/avalanche/evm-fetcher/sources/trader-joe-auto/TraderJoeAutoRequestHandlers.ts (hunk 2 above)
Comment (cehali) [reply packages/oracle-node/src/fetchers/evm-chain/avalanche/evm-fetcher/sources/trader-joe-auto/TraderJoeAutoRequestHandlers.ts:107]: Added

File: packages/oracle-node/test/fetchers/avalanche-evm.spec.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (koonopek) [reply packages/oracle-node/test/fetchers/avalanche-evm.spec.ts:409]: Is it usdc to avax ratio? then it should be 1/ ~12 = 0.083 (one decimal less)

**Review by koonopek — DISMISSED** [review]

**Comment by cehali** [comment]
@koonopek please reapprove 😃

**Pull request merged**